libs/sdk-core/src/swap_in/segwit.rs

```diff
@@ -0,0 +1,178 @@
+                        ]),
+                        ..utxo.try_into()?
+                    })
+                })
+                .collect::<Result<_, ReceiveSwapError>>()?,
+            output: vec![TxOut {
+                value: 0,
+                script_pubkey: destination_address.script_pubkey(),
+            }],
+        })
+    }
+
+    pub fn create_refund_tx(
+        &self,
+        swap_info: &SwapInfo,
+        utxos: &[SwapOutput],
+        destination_address: &Address,
+        sat_per_vbyte: u32,
+    ) -> ReceiveSwapResult<Transaction> {
+        let weight = self
+            .create_fake_refund_tx(swap_info, utxos, destination_address)?
+            .weight();
+        let fee = compute_tx_fee(weight, sat_per_vbyte);
+        let value: u64 = utxos
+            .iter()
+            .map(|utxo| utxo.amount_sat)
+            .sum::<u64>()
+            .saturating_sub(fee);
+        if value == 0 {
+            return Err(ReceiveSwapError::OutputValueBelowDust);
+        }
+
+        let lock_time = utxos.iter().fold(0, |accum, item| {
+            let confirmed_height = item.confirmed_at_height.unwrap_or(0);
+            if accum >= confirmed_height + (swap_info.lock_height as u32) {
+                accum
+            } else {
+                confirmed_height + (swap_info.lock_height as u32)
+            }
+        });
+
+        let input_script = create_submarine_swap_script(
+            &swap_info.payment_hash,
+            &swap_info.swapper_public_key,
+            &swap_info.public_key,
+            swap_info.lock_height,
+        )?;
+
+        let mut tx = Transaction {
+            version: 2,
+            lock_time: PackedLockTime(lock_time),
+            input: utxos
+                .iter()
+                .map(|utxo| utxo.try_into())
+                .collect::<Result<_, _>>()?,
+            output: vec![TxOut {
+                value,
+                script_pubkey: destination_address.script_pubkey(),
+            }],
+        };
+
+        let scpt = Secp256k1::signing_only();
+        let cloned_tx = tx.clone();
+        let mut signer = SighashCache::new(&cloned_tx);
+        for (input_index, input) in tx.input.iter_mut().enumerate() {
+            let sig = signer.segwit_signature_hash(
+                input_index,
+                &input_script,
+                utxos[input_index].amount_sat,
+                EcdsaSighashType::All,
+            )?;
+            let msg = Message::from_slice(&sig[..])?;
+            let secret_key = SecretKey::from_slice(&swap_info.private_key)?;
+            let sig = scpt.sign_ecdsa(&msg, &secret_key);
+
+            let mut sigvec = sig.serialize_der().to_vec();
+            sigvec.push(EcdsaSighashType::All as u8);
+
+            let witness: Vec<Vec<u8>> = vec![sigvec, vec![], serialize(&input_script)];
+
+            let w = Witness::from_vec(witness);
+            input.witness = w;
+        }
+
+        Ok(tx)
+    }
+
+    pub async fn get_swap_payment(
```

Comment:
Nit: this name makes it seem like we're getting some "payment" struct (like reading from db). Maybe `complete_swap_payment` or `complete_payment`?

Reply:
I went for `payout_swap`: 9fec848ab67f58da6bf6ec7678355509e28ab60c